core/baseline-core-v1.0-psd01.md

```diff
@@ -3001,60 +3003,227 @@ This section of the document focuses on the requirements and considerations rela
+
+Test Steps:
+
+1. One BPI workstep participants proposes a validation method for the external non-authoritative, non-deterministic input data to be used in the test workstep.
+2. Each BPI workstep participant verifies that the proposed validation method on sample input data produces reliable results.
+3. Each BPI workstep participant either accepts or rejetcs the proposal. 
+4. Once the validation method is agreed upon, it is recorded in the BPI.
+
+Test Passing Criteria:
+
+* The BPI workstep participants have agreed upon a validation method for the external non-authoritative, non-deterministic input data.
+* The agreed-upon validation method has been tested and shown to produce reliable results.
+* The validation method and testing results have been recorded on the test BPI.
+
+#### **[R317]** 
 A BPI workstep participant MUST be able to validate that the validation criteria of external non-authoritative,  non-deterministic input data to the BPI workstep has been met.
 
 *Appropriate data validation methods vary on a case-by-case basis depending upon the data types, sources and formats. As such, it is up to the BPI participants to agree upon the optimal validation method for their implementation. This includes design decisions such as what threshold of unresponsive oracles leads to a rejected input, how to aggregate the oracle responses and remove outliers, signing data to ensure provenance, implementing cryptographic mechanisms such as threshold signatures/secret sharing and TEEs such as Intel SGX or AMD SEV to obfuscate data from oracle providers.*
 
-[[R318]](#r318) Testability: Validation of non-authoritative, non-deterministic BPI input data is an extension of the state object validation requirement, [[R25]](#r25) which is testable, because non-authoritative, non-deterministic BPI input data sources will be part of a BPI state object. Therefore, the requirement is testable.
+[[R317]](#r317) Testability: 
+
+Preconditions:
+
+* A test BPI system is available and functional.
+* At least one source of external non-authoritative, non-deterministic input data.
+* A test workstep that takes external non-authoritative, non-deterministic input data.
+* The test workstep participant has access to the external non-authoritative, non-deterministic input data.
+* The validation criteria for the external non-authoritative, non-deterministic input data has been agreed upon and documented by the BPI workstep participants.
+* The validation criteria for the external non-authoritative, non-deterministic input data has been implemented in the test workstep.
+
+Test Steps:
+
+1. Input data from an external non-authoritative, non-deterministic source.
+2. Validate the external non-authoritative, non-deterministic input data against the agreed-upon validation criteria in the test workstep.
+3. Record the validation result along with the validation criteria and input data.
+
```

Comment:
Need a fourth step where another participant validates the recorded result using the validation criteria.

Reply:
added including test passing cirteria